presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -591,4 +622,26 @@ public Statement getStatement()
             return statement;
         }
     }
+
+    private boolean serverSupportsVariablePrecisionTimestamp()
+    {
+        return getNumericPrestoServerVersion() >= 335;
+    }
+
+    private boolean serverSupportsVariablePrecisionTimestampWithTimeZone()
+    {
+        return getNumericPrestoServerVersion() >= 337;
+    }
+
+    private int getNumericPrestoServerVersion()
+    {
+        try {
+            return Integer.parseInt(prestoServerVersion.get());
```

Comment:
i think it would be nice to validate our assumption that the version string matches
```
\d+(-SNAPSHOT)?
```

Then use max int for SNPASHOT case and
use parseInt for the other case.

This way we would catch when the version is sth else than we assume

Reply:
% the fact it can be "unknown" if Presto is run from IntelliJ.

https://github.com/prestosql/presto/blob/dd23a7b7db6cc1a2fe720075e10106f0b6230805/presto-server-main/src/main/java/io/prestosql/server/PrestoServer.java#L37-L38